AutomatedTesting/Gem/PythonTests/largeworlds/gradient_signal/EditorScripts/ImageGradient_ModifiesSurfaces.py

```diff
@@ -0,0 +1,119 @@
+    components_to_add = ["Image Gradient", "Gradient Transform Modifier", "Shape Reference",
+                         "Gradient Surface Tag Emitter"]
+    entity_position = math.Vector3(0.0, 0.0, 0.0)
+    new_entity_id = editor.ToolsApplicationRequestBus(
+        bus.Broadcast, "CreateNewEntityAtPosition", entity_position, EntityId.EntityId()
+    )
+    Report.critical_result(Tests.image_gradient_entity_created, new_entity_id.IsValid())
+    image_gradient_entity = entity.EditorEntity.create_editor_entity_at(entity_position, "Image Gradient")
+    image_gradient_entity.add_components(components_to_add)
+    test_img_gradient_path = os.path.join("Assets", "ImageGradients", "image_grad_test_gsi.png.streamingimage")
+    asset_id = asset.AssetCatalogRequestBus(bus.Broadcast, "GetAssetIdByPath", test_img_gradient_path, math.Uuid(),
+                                            False)
+    image_gradient_entity.components[0].set_component_property_value("Configuration|Image Asset", asset_id)
+    success = image_gradient_entity.components[0].get_component_property_value("Configuration|Image Asset") == asset_id
+    Report.result(Tests.image_gradient_assigned, success)
+
+    # 3) Create vegetation and planting surface entities, and assign the Image Gradient entity's Shape Reference
+
+    # Create vegetation entity
+    purple_flower_prefab_path = os.path.join("assets", "prefabs", "PurpleFlower.spawnable")
+    spawner_entity = dynveg.create_prefab_vegetation_area("Instance Spawner", entity_position, 50.0, 50.0, 10.0,
+                                                          purple_flower_prefab_path)
+    spawner_entity.add_component("Vegetation Surface Mask Filter")
+
+    # Create surface entity
+    dynveg.create_surface_entity("Box Shape", entity_position, 50.0, 50.0, 1.0)
+
+    # Assign Image Gradient entity's Shape Reference
+    image_gradient_entity.components[2].set_component_property_value("Configuration|Shape Entity Id", spawner_entity.id)
+
+    # 4) Assign surface tags to the required components
+    tag_list = [surface_data.SurfaceTag()]
+    surface_tags = {"terrain": 3363197873}
+
+    # Set the Veg Spawner entity's Surface Tag Mask Filter component to include the "terrain" tag
+    hydra.get_property_tree(spawner_entity.components[3])
+    hydra.get_set_test(spawner_entity, 3, "Configuration|Inclusion|Surface Tags", tag_list)
+    hydra.get_set_test(spawner_entity, 3, "Configuration|Inclusion|Surface Tags|[0]|Surface Tag",
+                       surface_tags["terrain"])
+
+    # Set the Image Gradient entity's Gradient Surface Tag Emitter component to modify the "terrain" tag
+    # NOTE: This requires a disable/re-enable of the component to force a refresh as assigning a tag via script does not
+    grad_surf_tag_emitter_component = image_gradient_entity.components[3]
+    grad_surf_tag_emitter_component.add_container_item("Configuration|Extended Tags", 0, surface_tags["terrain"])
```

Comment:
Then this would become (I think):
```
grad_surf_tag_emitter_component.add_container_item("Configuration|Extended Tags", 0, tag_list[0])
```

Also, shouldn't the need to force a refresh get entered as a bug?

Reply:
Yep, was waiting until this test was checked in for easy repro steps, but I suppose if no one is going to jump on it right away, I can log it up now: https://github.com/o3de/o3de/issues/7597